lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,6 +20,7 @@ def as_json
         private
 
         def type_for(serializer)
+          return serializer._type.call(serializer.object) if serializer._type.is_a?(Proc)
```

Comment:
is there a scenario in which a lambda / Proc / block wouldn't respond to call or be of type Proc?